toast-tk-runtime/src/main/java/io/toast/tk/runtime/result/AbstractResultHandler.java

```diff
@@ -20,24 +20,34 @@ public ITestResult result(T value, final String expected) {
 			return result;
 		}
 		
-		Object expectedValue = expected != null  && isVar(expected) ? 
-				objectRepository.getUserVariables().get(expected) : expected;
-		
+		Object expectedValue = null;
+		if(expected != null  && isVar(expected)) {
+			objectRepository.getUserVariables().get(expected);
+		} else {
+			if(expected != null) {
+				expectedValue = expected.trim();
+			} else {
+				expectedValue = expected;
+			}
+		}
+
 		if(value == null){
 			if(expectedValue != null){
 				markResultAsFailure(result);
 			}
 		}
-		else if(expectedValue != null && !value.toString().equals(expectedValue)){
+		else if(expectedValue != null && !expectedValue.equals("") && !value.toString().equals(expectedValue)
+					&& !shouldStoreResult(expected) // The comparison of stored result is not well executed
+													// If the user want to compare result, he has to use the phrase
+				){
 			markResultAsFailure(result);
 		}
 		return result;
     }
 
 	private boolean shouldStoreResult(String expected) {
-		return expected != null && 
-			isVar(expected) 
-			&& !objectRepository.getUserVariables().containsKey(expected);
+		return expected != null && isVar(expected);
+			//&& !objectRepository.getUserVariables().containsKey(expected);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)